pkg/virtualnetwork/conn.go

```diff
@@ -29,6 +33,9 @@ func (n *VirtualNetwork) DialContextTCP(ctx context.Context, addr string) (net.C
 		return nil, err
 	}
 
+	if port > math.MaxUint16 {
+		return nil, errors.New("invalid port number")
+	}
```

Comment:
This could/should be moved to `splitIPPort` instead of duplicating the check in every caller.